templates/asset_form.html.twig

```diff
@@ -367,6 +598,141 @@
                   ) }}
                {% endif %}
 
+               {% if item.isField('begin_date') %}
+                  {{ fields.dateField('begin_date', item.fields['begin_date'], __('Start date')) }}
+               {% endif %}
+
+               {% if item.getType == 'Contract' and item.isField('duration') %}
+                  {% if item.fields['begin_date'] is not empty and item.fields['duration'] is not empty %}
+                     {% set warranty_expiration %}
+                        -> {{ Infocom__getWarrantyExpir(
+                           item.fields['begin_date'],
+                           item.fields['duration'],
+                           0,
+                           true,
+                           item.fields['renewal'] == constant('Contract::RENEWAL_TACIT'))
+                        }}
+                     {% endset %}
+                  {% endif %}
+                  {{ fields.dropdownNumberField('duration', item.fields['duration'], __('Initial contract period'), {
+                     'rand': rand,
+                     'min': 1,
+                     'max': 120,
+                     'step': 1,
+                     'toadd': [constant('Dropdown::EMPTY_VALUE')],
+                     'unit': 'month',
+                     'add_field_html': warranty_expiration|default('')
+                  }) }}
+               {% endif %}
+
+               {% if item.isField('notice') %}
+                  {% if item.fields['begin_date'] is not empty and item.fields['notice'] > 0 and item.fields['duration'] is not empty %}
+                     {% set notice_info %}
+                        -> {{ Infocom__getWarrantyExpir(
+                        item.fields['begin_date'],
+                        item.fields['duration'],
+                        item.fields['notice'],
+                        true)
+                     }}
+                     {% endset %}
+                  {% endif %}
+                  {{ fields.dropdownNumberField('notice', item.fields['notice'], __('Notice'), {
+                     'rand': rand,
+                     'min': 0,
+                     'max': 120,
+                     'step': 1,
+                     'unit': 'month',
+                     'add_field_html': notice_info|default('')
+                  }) }}
+               {% endif %}
+
+               {% if item.isField('accounting_number') %}
+                  {{ fields.autoNameField(
+                     "accounting_number",
+                     item,
+                     __('Account number'),
+                     withtemplate
+                  ) }}
+               {% endif %}
+
+               {% if item.isField('periodicity') %}
```

Comment:
to precise with the above comments with `numberFields`, this field is a perfect usage of `dropdownNumberField` as we add textual options